### PR TITLE
Validate Space Names

### PR DESCRIPTION
**Pull request by kobergj**
We now return `BAD REQUEST` when space names are
-  too long (max 255 characters)
-  containing evil characters (`/`, `\`, `.`, `\\`, `:`, `?`, `*`, `"`, `>`, `<`, `|`)

Additionally leading and trailing spaces will be removed silently.

fixes this: https://github.com/owncloud/ocis/issues/4925

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiSpaces-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/16583/42/1) failed. Further test are cancelled...

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-8</strong>](https://drone.owncloud.com/owncloud/ocis/16583/54/1) failed. Further test are cancelled...

**Review by micbar — APPROVED** [review]
LGTM

File: services/graph/pkg/service/v0/drives.go [hunk 1]
```diff
@@ -33,6 +33,20 @@ import (
...
 )
 
+var (
+	_invalidSpaceNameCharacters = []string{`/`, `\`, `.`, `:`, `?`, `*`, `"`, `>`, `<`, `|`}
+	_maxSpaceNameLength         = 255
+
+	// ErrNameTooLong is thrown when the spacename is too long
+	ErrNameTooLong = fmt.Errorf("spacename must be smaller than %d", _maxSpaceNameLength)
+
+	// ErrNameEmpty is thrown when the spacename is too long
```
Comment (fschade) [inline services/graph/pkg/service/v0/drives.go:43]: … when the spacename is empty

Comment (micbar) [reply services/graph/pkg/service/v0/drives.go:43]:
```suggestion
	// ErrNameEmpty is thrown when the spacename is empty
```

**Review by fschade — APPROVED** [review]

**Comment by butonic** [comment]
https://drone.owncloud.com/owncloud/ocis/16604 is green, force merging

**Pull request merged**